client/my-sites/plans/current-plan/purchases-listing.jsx

```diff
@@ -0,0 +1,196 @@
+/**
+ * External dependencies
+ */
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import { get, invoke } from 'lodash';
+
+/**
+ * Internal Dependencies
+ */
+import {
+	getCurrentPlan,
+	isCurrentPlanExpiring,
+	isRequestingSitePlans,
+} from 'state/sites/plans/selectors';
+import { getSelectedSite, getSelectedSiteId } from 'state/ui/selectors';
+import Button from 'components/button';
+import Card from 'components/card';
+import MyPlanCard from './my-plan-card';
+import QuerySites from 'components/data/query-sites';
+import QuerySitePlans from 'components/data/query-site-plans';
+import QuerySitePurchases from 'components/data/query-site-purchases';
+import { getPlan } from 'lib/plans';
+import { getByPurchaseId } from 'state/purchases/selectors';
+import { isPartnerPurchase, shouldAddPaymentSourceInsteadOfRenewingNow } from 'lib/purchases';
+import { isFreeJetpackPlan, isFreePlan } from 'lib/products-values';
+import { managePurchase } from 'me/purchases/paths';
+
+class PurchasesListing extends Component {
+	static propTypes = {
+		currentPlan: PropTypes.object,
+		isPlanExpiring: PropTypes.bool,
+		isRequestingSitePlans: PropTypes.bool,
+		selectedSite: PropTypes.object,
+		selectedSiteId: PropTypes.number,
+		purchase: PropTypes.object,
+	};
+
+	isLoading() {
+		const { selectedSite, isRequestingSitePlans: isRequestingPlans } = this.props;
+
+		return ! selectedSite || isRequestingPlans;
+	}
+
+	isFreePlan() {
+		const { currentPlan } = this.props;
+
+		return ! currentPlan || isFreePlan( currentPlan ) || isFreeJetpackPlan( currentPlan );
+	}
+
+	getPlanObject() {
+		const { selectedSite } = this.props;
+		const currentPlanSlug = get( selectedSite, 'plan.product_slug' );
```

Comment:
Since we have a `getSitePlanSlug` selector, I wonder if there is any reason to not directly use that.